src/Suppressor.jl

```diff
@@ -36,7 +36,7 @@ macro suppress(block)
         finally
             if ccall(:jl_generating_output, Cint, ()) == 0
                 redirect_stdout(original_stdout)
-                close(out_wr)
+                flush(out_wr)
```

Comment:
My main open question here is:
- Does GC close a PipeEndpoint when it is GC'd? Or do we need to manually add a Finalizer to do this?

Reply:
The answer is yes it does.